### PR TITLE
Fixed #27899 -- Added phrase search query for postgres full text search

### PR DESCRIPTION
**Pull request by andriisoldatenko**
https://code.djangoproject.com/ticket/27899

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
Looks like related we have related PR https://github.com/django/django/pull/8313, I can update my PR

**Comment by andriisoldatenko** [comment]
@mjtamlyn please review.

File: tests/postgres_tests/test_search.py [hunk 1]
```diff
@@ -286,3 +286,20 @@ def test_ranking_chaining(self):
             rank=SearchRank(SearchVector('dialogue'), SearchQuery('brave sir robin')),
         ).filter(rank__gt=0.3)
         self.assertSequenceEqual(searched, [self.verse0])
+
+
+@skipIfDBFeature('has_phraseto_tsquery_support')
```
Comment (atombrella) [inline tests/postgres_tests/test_search.py]: Shouldn't this be `skipUnlessDBFeature`?

Comment (andriisoldatenko) [inline tests/postgres_tests/test_search.py]: yes thanks! I'll fix.

Comment (andriisoldatenko) [inline tests/postgres_tests/test_search.py]: @atombrella done.

**Comment by andriisoldatenko** [comment]
@timgraham Can you please help me what I'm doing wrong?
```
(django-dev) ➜  docs git:(ticket_27899) ✗ make spelling
sphinx-build -b spelling -n -d _build/doctrees -D language=   . _build/spelling
Running Sphinx v1.5.6
A Translator for the djangohtml builder is changed.
A Translator for the json builder is changed.
Initializing Spelling Checker
loading translations []... not available for built-in messages
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [spelling]: all documents
updating environment: 0 added, 0 changed, 0 removed
looking for now-outdated files... none found
preparing documents... done
writing output... [100%] topics/testing/tools
Spelling checker messages written to /Users/andrii/work/django/docs/_build/spelling/output.txt
build succeeded.

Check finished. Wrong words can be found in  _build/spelling/output.txt.
(django-dev) ➜  docs git:(ticket_27899) ✗ cat _build/spelling/output.txt
```

**Comment by timgraham** [comment]
The docs should build on your next update, see https://github.com/django/django/pull/8515#issuecomment-302394570.

Please check your patch using the [patch review checklist](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/submitting-patches/#patch-review-checklist). At a quick glance, I see some `versionadded` annotations are missing.

**Comment by andriisoldatenko** [comment]
@atombrella @timgraham updated according checklist.

File: docs/ref/contrib/postgres/search.txt [hunk 2]
```diff
@@ -87,6 +87,50 @@ looks for matches for all of the resulting terms.
...
+operators `&` (AND), `|` (OR), and `!` (NOT) as well as the phrase search operator
+`<->` (FOLLOWED BY) for better search results.
+
+    >>> from django.contrib.postgres.search import RawSearchQuery
+    >>> Entry.objects.annotate(
+    ...     search=SearchVector('body_text'),
+    ... ).filter(search=RawSearchQuery('toast <-> recipe'))
+    [<Entry: Cheese on Toast recipes>]
+
+
```
Comment (francoisfreitag) [inline docs/ref/contrib/postgres/search.txt]: There is usually just one line before a section header.

File: tests/postgres_tests/test_search.py [hunk 3]
```diff
@@ -286,3 +286,20 @@ def test_ranking_chaining(self):
...
+            dialogue__search=PhraseSearchQuery('body burned'))
+        self.assertSequenceEqual(searched, [self.verse1])
+
+
+@modify_settings(INSTALLED_APPS={'append': 'django.contrib.postgres'})
+class TestRawSearchQuery(GrailTestData, PostgreSQLTestCase):
+    def test_simple_raw_search(self):
+        searched = Line.objects.filter(
+            dialogue__search=RawSearchQuery('shall & use'))
+        self.assertSequenceEqual(searched, [self.bedemir0])
```
Comment (francoisfreitag) [inline tests/postgres_tests/test_search.py]: You might want to write some tests to prove that the new query search subclasses combine correctly with SearchQuery.

**Comment by timgraham** [comment]
Continued in https://github.com/django/django/pull/10359.

**Pull request closed without merging**